### PR TITLE
re-organize install guide and add more details

### PR DESCRIPTION
**Pull request by BenTheElder**
fixes https://github.com/kubernetes-sigs/kind/issues/2533

### CODE REVIEW

File: site/content/docs/user/quick-start.md [hunk 1]
```diff
@@ -19,33 +19,16 @@ description: |-
...
 
-Stable binaries are also available on the [releases] page.
-Stable releases are generally recommended for CI usage in particular.
-To install, download the binary for your platform from "Assets" and place this
-into your `$PATH`.
+Stable tagged releases (currently {{< stableVersion >}}) are generally strongly recommended for CI usage in particular.
 
-On Linux:
+You may need to install the latest code from source at HEAD if you are developing Kubernetes itself at HEAD / the latest sources.
+### Installing With A Package Manager
```
Comment (BenTheElder) [inline site/content/docs/user/quick-start.md:29]: nit: missing blank line

File: site/content/docs/user/quick-start.md [hunk 2]
```diff
@@ -59,25 +42,78 @@ On macOS via MacPorts:
...
 {{< /codeFromInline >}}
 
-On Windows via Chocolatey (https://chocolatey.org/packages/kind)
-{{< codeFromInline lang="powershell" >}}
-choco install kind
-{{< /codeFromInline >}}
+### Installing From Source
+
+In addition to the pre-built binary + package manager installation options listed
+above you can install from kind from source with `GO111MODULE="on" go get sigs.k8s.io/kind@{{< stableVersion >}}` or clone this repo
```
Comment (tao12345666333) [inline site/content/docs/user/quick-start.md:83]:
```suggestion
above you can install kind from source with `GO111MODULE="on" go get sigs.k8s.io/kind@{{< stableVersion >}}` or clone this repo
```

Comment (AkihiroSuda) [inline site/content/docs/user/quick-start.md:83]: `go install` is more preferable for recent versions of Go
https://github.com/golang/go/issues/43684

Comment (BenTheElder) [reply site/content/docs/user/quick-start.md:83]: There are notes about that just a few sentences below in the fully detailed section on installing from source with go, currently this command supports more versions however.

**Comment by aojea** [comment]
why is this WIP¿?

**Comment by BenTheElder** [comment]
outstanding comments I didn't get back to

File: site/content/docs/user/quick-start.md (hunk 2 above)
Comment (BenTheElder) [reply site/content/docs/user/quick-start.md]: done

File: site/content/docs/user/quick-start.md (hunk 1 above)
Comment (BenTheElder) [reply site/content/docs/user/quick-start.md:30]: done.

**Pull request merged**